src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Main {
+    private static final FigureSupplier figureSupplier = new FigureSupplier();
```

Comment:
Avoid using static for the `figureSupplier` instance. Instead, create an instance of `FigureSupplier` inside the `main` method. Static methods and fields are generally discouraged as they can lead to issues with object-oriented design principles.